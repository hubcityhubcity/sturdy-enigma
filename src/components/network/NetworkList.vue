<template>
  <ul class="network-list">
    <li
      v-for="network in networks"
      :key="network.chain_id"
      class="select-network-item"
      :class="{ selected: currentNetwork === network.id }"
      @click="network.chain_id ? selectNetworkHandler(network) : false"
    >
      <NetworkItem :networkitem="network" />
    </li>
  </ul>
</template>

<script>
import { mapGetters } from "vuex"
import NetworkItem from "./NetworkItem"

export default {
  name: `page-network`,
  components: {
    NetworkItem
  },
  props: {
    networks: {
      type: Array,
      required: true
    }
  },
  computed: {
    ...mapGetters([`network`]),
    currentNetwork() {
      return this.network
    }
  },
  methods: {
    async selectNetworkHandler(network) {
      if (this.network !== network.id) {
        this.$store.dispatch(`setNetwork`, network)
      }
    }
  }
}
</script>
<style scoped>
.network-list {
  margin-bottom: 4rem;
}
</style>
