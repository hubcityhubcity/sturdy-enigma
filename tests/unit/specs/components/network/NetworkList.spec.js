import { shallowMount, createLocalVue } from "@vue/test-utils"
import NetworkList from "network/NetworkList"

const localVue = createLocalVue()

describe(`NetworkList`, () => {
  let wrapper

  const networks = [
    {
      id: "gaia-testnet",
      chain_id: "gaia-123",
      logo_url: "cosmos-logo.png",
      testnet: true,
      title: "Cosmos Hub Test"
    },
    {
      id: "cosmoshub",
      chain_id: "cosmoshub",
      logo_url: "cosmos-logo.png",
      testnet: false,
      title: "Cosmos Hub"
    }
  ]

  beforeEach(() => {
    wrapper = shallowMount(NetworkList, {
      localVue,
      propsData: {
        networks
      },
      mocks: {
        $store: {
          dispatch: jest.fn(),
          state: {
            connection: {
              network: `cosmoshub`
            },
            session: {
              signedIn: false
            }
          }
        },
        $route: {
          params: { height: `100` }
        },
        $router: {
          push: jest.fn()
        }
      },
      stubs: [`router-link`]
    })
  })

  it(`shows a list of networks`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it("sets new network when clicking list item", () => {
    wrapper.find(".select-network-item:not(.selected)").trigger("click")
    expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith(`setNetwork`, {
      id: "gaia-testnet",
      chain_id: "gaia-123",
      logo_url: "cosmos-logo.png",
      testnet: true,
      title: "Cosmos Hub Test"
    })
  })

  it("does not change network when already selected", () => {
    wrapper.find(".select-network-item.selected").trigger("click")
    expect(wrapper.vm.$store.dispatch).not.toHaveBeenCalledWith()
  })
})
