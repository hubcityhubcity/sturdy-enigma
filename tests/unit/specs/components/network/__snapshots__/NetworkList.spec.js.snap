// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkList shows a list of networks 1`] = `
<ul
  class="network-list"
>
  <li
    class="select-network-item"
  >
    <networkitem-stub
      networkitem="[object Object]"
    />
  </li>
  <li
    class="select-network-item selected"
  >
    <networkitem-stub
      networkitem="[object Object]"
    />
  </li>
</ul>
`;
